clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/CachingKafkaProducerFactory.java

```diff
@@ -54,17 +54,76 @@
     /**
      * Creates a new producer factory.
      * <p>
-     * Use {@link KafkaProducerFactory#sharedProducerFactory(Vertx)} to create producers that can safely be shared
-     * between verticle instances.
+     * Use {@link #sharedFactory(Vertx)} to create producers that can safely be shared between verticle instances.
      *
      * @param producerInstanceSupplier The function that provides new producer instances. Parameters are the producer
-     *                                 name and the producer configuration.
+     *            name and the producer configuration.
      */
-    public CachingKafkaProducerFactory(
+    private CachingKafkaProducerFactory(
             final BiFunction<String, Map<String, String>, KafkaProducer<K, V>> producerInstanceSupplier) {
         this.producerInstanceSupplier = producerInstanceSupplier;
     }
 
+    /**
+     * Creates a new factory that produces {@link KafkaProducer#createShared(Vertx, String, Map) sharedFactory
+     * producers}. Shared producers can safely be sharedFactory between verticle instances and improve efficiency by
+     * leveraging the batching capabilities of the Kafka client.
+     *
+     * Producers with the same name will be shared (per Vert.x instance), meaning they are shared between instances of
+     * the factory.
+     * <p>
+     * Config must always be the same for the same key in
+     * {@link #getOrCreateProducer(String, KafkaProducerConfigProperties)}.
+     * <p>
```

Comment:
How about:
```
Instances returned here must use the same config in 
{@link #getOrCreateProducer(String, KafkaProducerConfigProperties)}
invocations that use the same key.
```
?

Reply:
I think the whole sentence should be removed. This is about the `getOrCreateProducer` method where the JavaDoc already states "The given config is ignored when an existing producer is returned". It always applies to this method regardless of the producer instance supplier.